### PR TITLE
Add DataFrame.describe

### PR DESCRIPTION
**Pull request by sinhrks**
```
import numpy as np
import pandas as pd
import dask.dataframe as dd
df = pd.DataFrame({'a': range(20) * 4,
                   'b': range(4) * 20})
ddf = dd.from_pandas(df, 4)
ddf.describe().compute()
#                a          b
# count  80.000000  80.000000
# mean    9.500000   1.500000
# std     5.802662   1.125088
# min     0.000000   0.000000
# 25%     4.750000   0.750000
# 50%     9.500000   1.500000
# 75%    14.250000   2.250000
# max    19.000000   3.000000
```

Also, fixed ``_get_numeric_data`` and ``dtypes`` to use same cached result.

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -626,6 +626,27 @@ def quantile(self, q=0.5, axis=0):
...
+    def describe(self):
+        # currently, only numeric describe is supported
+        num = self._get_numeric_data()
+
+        name = 'describe--' + tokenize(self)
+        stat_index = (['count', 'mean', 'std', 'min',
+                       '25%', '50%', '75%', 'max'])
+        stats = [num.count(), num.mean(), num.std(), num.min(),
+                 num.quantile(0.25), num.quantile(0.5), num.quantile(0.75),
+                 num.max()]
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Should we compute the quantiles all in one operation?

**Comment by mrocklin** [comment]
This is really cool

Fixes https://github.com/blaze/dask/issues/333

File: dask/dataframe/core.py (hunk 1 above)
Comment (sinhrks) [inline dask/dataframe/core.py]: Correct, the change should simplify the internal computation.

**Comment by sinhrks** [comment]
Fixed. Expect it will be useful for demonstration:)

index

**Comment by jcrist** [comment]
This looks great (graph will definitely be good for demos). +1.

**Pull request merged**